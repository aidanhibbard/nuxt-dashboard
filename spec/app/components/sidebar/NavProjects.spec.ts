import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import NavProjects from '@/components/navigation/sidebar/NavProjects.vue'


describe('components/navigation/sidebar/NavProjects.vue', () => {
	it('renders project items and a More item', () => {
		const wrapper = mount(NavProjects, {
			props: {
				projects: [
					{ name: 'Design Engineering', url: '#', icon: { name: 'IconA' } as any },
					{ name: 'Sales & Marketing', url: '#', icon: { name: 'IconB' } as any },
				],
			},
		})

		expect(wrapper.text()).toContain('Projects')
		expect(wrapper.text()).toContain('Design Engineering')
		expect(wrapper.text()).toContain('Sales & Marketing')
		expect(wrapper.text()).toContain('More')
	})

	it('matches snapshot (desktop)', () => {
		;(globalThis as any).__sidebarIsMobile = false
		const wrapper = mount(NavProjects, {
			props: { projects: [
				{ name: 'Design Engineering', url: '#', icon: { name: 'IconA' } as any },
				{ name: 'Sales & Marketing', url: '#', icon: { name: 'IconB' } as any },
			] },
		})
		expect(wrapper.html()).toMatchSnapshot()
	})

	it('updates dropdown alignment based on mobile state', async () => {
		;(globalThis as any).__sidebarIsMobile = true
		const wrapper = mount(NavProjects, {
			props: { projects: [ { name: 'Design Engineering', url: '#', icon: { name: 'IconA' } as any } ] },
		})
		expect(wrapper.find('[data-side="bottom"]').exists()).toBe(true)
		expect(wrapper.find('[data-align="end"]').exists()).toBe(true)
	})
})


