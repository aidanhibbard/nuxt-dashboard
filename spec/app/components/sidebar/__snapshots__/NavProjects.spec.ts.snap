// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/navigation/sidebar/NavProjects.vue > matches snapshot (desktop) 1`] = `
"<div class="group-data-[collapsible=icon]:hidden">
  <div>Projects</div>
  <ul>
    <li><button as-child=""><a href="#">
          <!----><span>Design Engineering</span>
        </a></button>
      <div>
        <div as-child=""><button show-on-hover=""><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-ellipsis-icon lucide-ellipsis">
              <circle cx="12" cy="12" r="1"></circle>
              <circle cx="19" cy="12" r="1"></circle>
              <circle cx="5" cy="12" r="1"></circle>
            </svg><span class="sr-only">More</span></button></div>
        <div data-side="right" data-align="start" class="w-48 rounded-lg">
          <div class="dropdown-menu-item"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide text-muted-foreground lucide-folder-icon lucide-folder text-muted-foreground">
              <path d="M20 20a2 2 0 0 0 2-2V8a2 2 0 0 0-2-2h-7.9a2 2 0 0 1-1.69-.9L9.6 3.9A2 2 0 0 0 7.93 3H4a2 2 0 0 0-2 2v13a2 2 0 0 0 2 2Z"></path>
            </svg><span>View Project</span></div>
          <div class="dropdown-menu-item"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide text-muted-foreground lucide-forward-icon lucide-forward text-muted-foreground">
              <path d="m15 17 5-5-5-5"></path>
              <path d="M4 18v-2a4 4 0 0 1 4-4h12"></path>
            </svg><span>Share Project</span></div>
          <hr>
          <div class="dropdown-menu-item"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide text-muted-foreground lucide-trash2-icon lucide-trash-2 text-muted-foreground">
              <path d="M10 11v6"></path>
              <path d="M14 11v6"></path>
              <path d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6"></path>
              <path d="M3 6h18"></path>
              <path d="M8 6V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"></path>
            </svg><span>Delete Project</span></div>
        </div>
      </div>
    </li>
    <li><button as-child=""><a href="#">
          <!----><span>Sales &amp; Marketing</span>
        </a></button>
      <div>
        <div as-child=""><button show-on-hover=""><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-ellipsis-icon lucide-ellipsis">
              <circle cx="12" cy="12" r="1"></circle>
              <circle cx="19" cy="12" r="1"></circle>
              <circle cx="5" cy="12" r="1"></circle>
            </svg><span class="sr-only">More</span></button></div>
        <div data-side="right" data-align="start" class="w-48 rounded-lg">
          <div class="dropdown-menu-item"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide text-muted-foreground lucide-folder-icon lucide-folder text-muted-foreground">
              <path d="M20 20a2 2 0 0 0 2-2V8a2 2 0 0 0-2-2h-7.9a2 2 0 0 1-1.69-.9L9.6 3.9A2 2 0 0 0 7.93 3H4a2 2 0 0 0-2 2v13a2 2 0 0 0 2 2Z"></path>
            </svg><span>View Project</span></div>
          <div class="dropdown-menu-item"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide text-muted-foreground lucide-forward-icon lucide-forward text-muted-foreground">
              <path d="m15 17 5-5-5-5"></path>
              <path d="M4 18v-2a4 4 0 0 1 4-4h12"></path>
            </svg><span>Share Project</span></div>
          <hr>
          <div class="dropdown-menu-item"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide text-muted-foreground lucide-trash2-icon lucide-trash-2 text-muted-foreground">
              <path d="M10 11v6"></path>
              <path d="M14 11v6"></path>
              <path d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6"></path>
              <path d="M3 6h18"></path>
              <path d="M8 6V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"></path>
            </svg><span>Delete Project</span></div>
        </div>
      </div>
    </li>
    <li><button class="text-sidebar-foreground/70"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide text-sidebar-foreground/70 lucide-ellipsis-icon lucide-ellipsis text-sidebar-foreground/70">
          <circle cx="12" cy="12" r="1"></circle>
          <circle cx="19" cy="12" r="1"></circle>
          <circle cx="5" cy="12" r="1"></circle>
        </svg><span>More</span></button></li>
  </ul>
</div>"
`;
